README.md

```diff
@@ -1,10 +1,130 @@
-# steps-sign-apk
+# Android Sign ![Bitrise Build Status](https://app.bitrise.io/app/3b968e65d584db2a.svg?token=Yk1LUEjLZtIjeIW4OOZvKw&branch=master) [![Bitrise Step Version](https://shields.io/github/v/release/bitrise-steplib/steps-sign-apk?include_prereleases)](https://github.com/bitrise-steplib/steps-sign-apk/releases) ![GitHub License](https://img.shields.io/badge/license-MIT-lightgrey.svg) [![Bitrise Community](https://img.shields.io/badge/community-Bitrise%20Discuss-lightgrey)](https://discuss.bitrise.io/)
 
-## How to use this Step
+Signs your APK or Android App Bundle before uploading it to Google Play Store.
 
-Can be run directly with the [bitrise CLI](https://github.com/bitrise-io/bitrise),
-just `git clone` this repository, `cd` into it's folder in your Terminal/Command Line
-and call `bitrise run test`.
+Once you have uploaded your keystore file and provided your keystore credentials on the **Code Signing** tab of the Workflow Editor, the **Android Sign** Step signs your APK digitally. 
+Bitrise assigns Environment Variables to the uploaded file and credentials, and uses those in the respective fields of the **Android Sign** Step. 
+Once the Step runs, it produces a signed APK or App Bundle which will be used as the input value of the **App file path** field in the **Google Play Deploy** Step.   
 
-*Check the `bitrise.yml` file for required inputs which have to be
-added to your `.bitrise.secrets.yml` file!*
+### Configuring the Step
+
+1. Add the **Android Sign** Step after a build Step in your deploy workflow.
+2. Upload the keystore file to the **Upload file** field on the **Code Signing** tab.
+3. Provide your keystore password, keystore alias and private key password to the relevant fields on the **Code Signing** tab.
+4. Run your build.
+
+
+### Troubleshooting
+Make sure you have the **Android Sign** Step right after a build Steps but before **Deploy to Google Play** Step in your deploy workflow.
+If you wish to get your Android project signed automatically, use the **Android Sign** Step and do not set any gradle task for the signing, otherwise, the Step will fail.
+
+
+### Useful links
+- [Android code signing using Android Sign Step](https://devcenter.bitrise.io/code-signing/android-code-signing/android-code-signing-using-bitrise-sign-apk-step/)
+- [Android deployment](https://devcenter.bitrise.io/deploy/android-deploy/android-deployment-index/)
+
+
+### Related Steps
+- [Android Build](https://www.bitrise.io/integrations/steps/android-build)
+- [Gradle Runner](https://www.bitrise.io/integrations/steps/gradle-runner)
+- [Deploy to Bitrise.io](https://www.bitrise.io/integrations/steps/deploy-to-bitrise-io)
+
+## Examples
+
+### Build a bundle and sign it
+
+```
```

Comment:
What about separating these 3 phases with some title too:


Build an Android App Bundle
Codesign the Bundle
Deploy to the Bitrise build

Reply:
updated